lib/haml/railtie.rb

```diff
@@ -10,13 +10,32 @@
 module Haml
   class Railtie < ::Rails::Railtie
     initializer :haml do |app|
-      require "haml/template"
-      if defined?(::Sass::Rails::SassTemplate) && app.config.assets.enabled
-        require "haml/sass_rails_filter"
+      if defined?(ActiveSupport)
+        ActiveSupport.on_load(:before_initialize) do
```

Comment:
While I'm not sure whether we should use `before_initialize` or not, there is no need to configure before_initialize hook inside initializer, isn't it? I prefer something like:

```rb
module Haml
  class Railtie < ::Rails::Railtie
    config.before_initialize do
      # ...
    end
  end
end
```